### ISSUE TITLE
Loaded map tiles don't show and are not blocked by any filter

### ISSUE THREAD

**Issue by szymone** [uBlockOrigin/uBlock-issues#588]
### Prerequisites

- [ ] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [ ] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [ ] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [ ] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [ ] I am running the latest version of uBlock Origin
- [ ] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

This is not a filter issue, I've checked this in logger window. Please read steps to reproduce. The problem disappears after resizing the window, so no filters are involved. It seams that the uBlock extension influence the page behavior somehow. Map tiles requests are not blocked by any filter and all tiles are loaded from the network, but after that only some of them are displayed. Disabling uBlock resolves the issue.

uBlock extension with default settings causes site https://mapa-turystyczna.pl not working correctly. Only some map tiles are loaded and this is not a filter issue. Disabling uBlock extension resolves the issue.

### A specific URL where the issue occurs

https://mapa-turystyczna.pl

### Steps to Reproduce

1. Open https://mapa-turystyczna.pl page
2. Result: only a few map tiles are loaded
3. Resize browser window.
4. All tiles are loaded.

### Expected behavior:

All tiles should be loaded just after entering the site.

### Actual behavior:

Only single map tiles are loaded.

### Your environment

* uBlock Origin version: uBlock Origin v1.19.2
* Browser Name and version: Firefox 66.0.5 (64 bity)
* Operating System and version: MacOS Mojave 10.14.4

**Comment by uBlock-user** [comment]
@okiehsch

**Comment by gorhill** [comment]
Unable to reproduce, all load fine here -- including with POL filter lists.

Try to reproduce again after the steps under _"I tried to reproduce the issue when"_.

**Comment by uBlock-user** [comment]
> Only some map tiles are loaded 

could be lazy loading in Firefox..

**Comment by mapx-** [comment]
It seems to be a filter issue (`POL: Oficjalne Polskie Filtry do AdBlocka, uBlocka Origin i AdGuarda`)
tested in FF
filter: `pl##[class$="-ads"]`

to get a correct screen:
`mapa-turystyczna.pl#@#[class$="-ads"]`

**Comment by uBlock-user** [comment]
@mapx- Can you reproduce the issue here ?

**Comment by gorhill** [comment]
> seems to be a filter issue

But I would be able to reproduce? I can't and the only things blocked as per logger is google analytic and google tag services.

**Comment by uBlock-user** [comment]
Similar issue with the same symptons https://github.com/Leaflet/Leaflet/issues/694

**Comment by gorhill** [comment]
Hmm actually when I panned down the map I get blank tiles.

**Comment by mapx-** [comment]
![image](https://user-images.githubusercontent.com/10303732/58100774-97b9c080-7bde-11e9-806a-37281910f0c1.png)

**Comment by gorhill** [comment]
I get the issue I described above without uBO, so not a uBO issue.

**Comment by uBlock-user** [comment]
On my end, the map loads initially and then it lazy loads every direction I move in to,

**Comment by mapx-** [comment]
No issue for me disabling uBo.

**Comment by gorhill** [comment]
Weird, it's as if there are two distinct issue with same behavior? I get the blank tiles without uBO when I pan.

**Comment by mapx-** [comment]
Let's see if @MajkiIT can reproduce the issue

**Comment by mapx-** [comment]
Without the exception I get only the first tile (top left)

**Comment by uBlock-user** [comment]
Seems to be a site issue, it makes use of Leaflet js lib -- https://github.com/Leaflet/Leaflet/issues/694

**Comment by gorhill** [comment]
This matches pretty well OP's description, plus the fact that I can reproduce without uBO.

I found that clearing cookies for that site help modify the reproducibility.

**Comment by uBlock-user** [comment]
Disabling uBO and uMatrix altogether, I found that websites inserted some googleadserving cookies, they've probably employed anti adblock which produces white blank tiles upon detection.

**Comment by gorhill** [comment]
Gave another try, and after clearing cookies I got what @mapx- observed -- the look is different that what I described before -- there is just a tiny tile top-left corner. So that would be a (difficult to diagnose to be fair) filter issue.

**Comment by gorhill** [comment]
Also, I think the site detect when the browser console is opened through viewport size changes, hence why resizing fix the issue. The site creating a cookie to track whether or not some of the DOM elements were hidden. Probably best to use a filter such as `||mapa-turystyczna.pl^$generichide`.

**Comment by szymone** [comment]
Thanks you all for comments and checking the issue. Addressing your comments, there is no anti adblock functions on this site. Also the Leaflet issue you referred to is quite old and we're using newer version of Leaflet with this issue fixed. I wasn't able to reproduce this issue with uBlock disabled so this was the reason I've reported this issue here. I'm an author of this page and some users reported an issue to me that map is not working properly and disabling uBlock resolved the issue. Resizing window / opening console resolves this issue as its trigger resize event to Leaflet. However when the site is loaded time, map should appear correctly.

**Comment by szymone** [comment]
@gorhill In what browser were you able to reproduce this issue with uBlock disabled?

**Comment by gorhill** [comment]
Every time the site fails to tile properly, there is this error in the console:

    11:40:54.801 jRange : no width found, returning             249363b.js:1:376114

The exception filter seems to prevent this error from occurring, so this confirms a filter issue, the site minds elements is being hidden as a result of a POL filter. You will have to sort out the issue with POL filter list maintainer, uBO itself is not causing this issue.

**Comment by gorhill** [comment]
I reproduced using Firefox stable -- as per your opening comment information.

**Comment by mapx-** [comment]
@szymone  just try the filter indicated above

**Comment by szymone** [comment]
@gorhill Thanks for your answer. The comment from jRange is another issue, thanks for finding it. In general there is some issue with calculating DOM elements width, that results in jRange error and issues with displaying tiles. This issue is resolved by disabling the filter you've mentioned, however I'm not sure this is the direct cause. Applying this filter probably removes some element on the site which can cause issues with calculating widths. 
Also does the screenshot you've sent shows the reproduced issue with uBlock disabled? On this screen all tiles are loaded correctly, the map just end outside Polish borders. Not loaded tiles are gray.

**Comment by szymone** [comment]
@mapx- What does this filter #@#[class$="-ads"] exactly do to matched DOM element? Does it removes it from DOM, hides it or something else? We have a div element with "ts-ads" class and applying this filter modifies this div somehow and this causes the issue with calculating DOM elements width, resulting in loading tiles and jRange problems.

**Comment by mapx-** [comment]
add only
`mapa-turystyczna.pl#@#[class$="-ads"]`

and see if works

**Comment by uBlock-user** [comment]
>What does this filter #@#[class$="-ads"] exactly do

#@# are exception filters.

**Comment by szymone** [comment]
@mapx- @uBlock-user I know I can add this filter and it resolves the issue, but I can't tell this to every user accessing my site with uBlock enabled to do this. I was asking what exactly happens to DOM element (what is done by uBlock extension) when the filter ##[class$="-ads"] is applied.

**Comment by szymone** [comment]
I've just checked and calculating body element width with uBlock enabled with ##[class$="-ads"] filter returns 100 px. With uBlock or the filter disabled the correct value is calculated and then all tiles are loaded correctly and jRange issue doesn't occur. So please explain what happens to the site DOM when this filter is applied.

**Comment by mapx-** [comment]
the class is not removed anymore so if it's about your site you can simply rename that class.

**Comment by MajkiIT** [comment]
@szymone @gorhill @mapx- Thanks
https://github.com/MajkiIT/polish-ads-filter/commit/c271c2cbc1b32619b6534284b2991472ebe75faa#diff-711027ed0794992ae3b8710d874f3df7
